pkgs/development/python-modules/grpc-stubs/default.nix

```diff
@@ -0,0 +1,60 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  grpcio,
+  pythonOlder,
+  pytest,
+  pytestCheckHook,
+  pytest-mypy-plugins,
+  types-protobuf,
+  cffi,
+  filelock,
+  hatchling,
+  poetry-core,
+  setuptools,
+}:
+
+buildPythonPackage rec {
+  pname = "grpc-stubs";
+  version = "1.53.0.5";
+  format = "pyproject";
+
+  src = fetchFromGitHub {
+    owner = "shabbyrobe";
+    repo = "${pname}";
+    rev = "${version}";
+    hash = "sha256-an7xztaCqxOEmf74Rgb8q9u/WsojFYkBiwtLRa1qqBQ=";
+  };
+
+  disabled = pythonOlder "3.6";
+
+  dependencies = [
+    pytest
+    types-protobuf
+    cffi
+    filelock
+    hatchling
+    setuptools
+  ];
+
+  nativeBuildInputs = [
+    poetry-core
+  ];
```

Comment:
You want to move `setuptools`, `hatchling` & `poetry-core` to `build-system`, and maybe check if all 3 of them are used